src/abstract class.java

```diff
@@ -0,0 +1,76 @@
+import components.map.Map;
+import components.map.Map1L;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.logging.Logger;
+import java.util.logging.Level;
+
+public abstract class TrackerSecondary implements Tracker {
+    //logging to replace outprint
+    private static final Logger logger = Logger.getLogger(TrackerSecondary.class.getName());
+
+    protected Map<String, Double> trackMap;
+
+    public TrackerSecondary() {
```

Comment:
There should be no constructors or fields here. The abstract class is meant to be abstract. Your methods should only use kernel methods.